libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/FreeDriveLocationUpdater.kt

```diff
@@ -23,12 +23,21 @@ internal class FreeDriveLocationUpdater(
     private val navigationEventDispatcher: NavigationEventDispatcher,
     private val mapboxNavigator: MapboxNavigator,
     private val offlineNavigator: OfflineNavigator,
-    private val executorService: ScheduledExecutorService
+    private val executorService: ScheduledExecutorService,
+    private val electronicHorizonRequestBuilder: ElectronicHorizonRequestBuilder
 ) {
     private val callback = CurrentLocationEngineCallback(this)
     private var future: ScheduledFuture<*>? = null
+    private var electronicHorizonFuture: ScheduledFuture<*>? = null
     private var rawLocation: Location? = null
     private val handler = Handler(Looper.getMainLooper())
+    private val cachedLocations = mutableListOf<Location>()
+
+    companion object {
```

Comment:
Could these be parameterized as options so we could change the values in the call-site?

Reply:
done